src/Config.hs

```diff
@@ -332,3 +337,42 @@ readValueFromEnvWithDefault :: (Read a, OptParse.HasValue f)
 readValueFromEnvWithDefault key defVal envVars
   =  value defVal
   <> readValueFromEnv key envVars
+
+-- | Search for environment files in default locations and load them in order.
+--
+-- This function tries to read the following files in order to obtain
+-- environment configuration. This is implicit behavior and allows the user to
+-- configure vaultenv without setting up environment variables or passing CLI
+-- flags. This is nicer for interactive usage.
+readConfigFromEnvFiles :: IO [(String, String)]
+readConfigFromEnvFiles = do
+  xdgDir <- (Just <$> Dir.getXdgDirectory Dir.XdgConfig "vaultenv")
+    `catchIOError` (const $ pure Nothing)
+  cwd <- Dir.getCurrentDirectory
+  let
+    machineConfigFile = "/etc/vaultenv.conf"
+    userConfigFile :: Maybe FilePath
+    userConfigFile = fmap (++ "/config.conf") xdgDir
+    cwdConfigFile = cwd ++ "/.env"
+
+  -- @doesFileExist@ doesn't throw exceptions, it catches @IOError@s and
+  -- returns @False@ if those are encountered.
+  loadMachineConfig <- Dir.doesFileExist machineConfigFile
+  loadUserConfig <- case userConfigFile of
+     Nothing -> pure False
+     Just fp -> Dir.doesFileExist fp
+  loadCwdConfig <- Dir.doesFileExist cwdConfigFile
+
+  machineConfig <- if loadMachineConfig
+    then DotEnv.parseFile machineConfigFile
+    else pure []
+  userConfig <- if loadUserConfig
+    then DotEnv.parseFile (fromJust userConfigFile) -- safe because of loadUserConfig
```

Comment:
Nit: In theory there is a race condition here (the file could have been removed inbetween the time you checked that it exists and the time of reading it), but I think we're not going to care about this here.

Reply:
Yeah that's true, same for the other ones. Don't think we'll ever see it though